File: erpx_hrm/www/hr/js/grp_settings.js
var current_role_name = '';
function openEditPeople(role_name){
	current_role_name = role_name;
	var dataArray1 = [];
	var roleUsers = role_list.find((r) => { return r.name === role_name }).user_list;
	
	$('#transfer1').html('');
	var settings1 = {
		"dataArray": dataArray1,
		"itemName": "name",
		"valueName": "value",
		"rightTabNameText": "All People",
		"tabNameText": "People belong to this Role",
		"callable": function (items) {
			//console.dir(items)
		}
	};
	settings1.tabNameText = "People belong to " + role_name;
	$.each( all_user, function( key, val ) {
		if($.inArray( val, roleUsers )  !== -1)
			dataArray1.push({
				name: val,
				value: val,				
			});
		else
			dataArray1.push({
				name: val,
				value: val,
				selected: true,				
			});
	});	
	$("#transfer1").transfer(settings1);
}
function saveEditPeople(){
	arr_all_user = [];
	arr_user = [];
	$('#transfer1 .transfer-double-content-right li:not(.selected-hidden) input[type="checkbox"]').each(function(){
		arr_all_user.push($(this).val());
	});
	$('#transfer1 .transfer-double-content-left li:not(.selected-hidden) input[type="checkbox"]').each(function(){
		arr_user.push($(this).val());
	});
	update_role_users(arr_user, arr_all_user, current_role_name);
	$('#modal-epml-ppl').modal('close');
}
$(document).ready(function () {
	
});
var update_role_users = function(arr_user, arr_all_user, role_name){	
	frappe.call({
		method: "erpx_hrm.utils.user.add_role_from_array",
		args : {
			arr_user : arr_user,
			arr_all_user: arr_all_user,
			role_name: role_name
		},
		callback: function(r) {
			location.reload();
		}
	});
}

//Edit permission
var permission_fields = [
	"Basic Information",
	"All About Me",
	"Certificate and License",
	"Compensation and Bank Information",
	"Compensation History",
	"Disciplinary Action",
	"Emergency Contact",
	"Family Information",
	"Personal Information",
	"Qualifications"
];
function openEditPermission(role_name){
	$('#modal-empl-per #sRoleName').html(role_name);
	$('#modal-empl-per #per_role_name').val(role_name);		

	$.each(permission_fields, function (key, element) {	
		$(`#modal-empl-per [data-fieldvalue="${element}"]`).attr('old_name', '');
		$(`#modal-empl-per [data-fieldvalue="${element}"]`).val('').formSelect();
	});
	frappe.call({
		method: "frappe.client.get_list",
		args: {
			doctype: "Section Permission",
			fields: ["*"],
			filters: [["role", "=", role_name]],
			limit_page_length: 0
		},
		callback: function (r) {
			r.message.forEach(function(element, index){
				$(`#modal-empl-per [data-fieldvalue="${element.section}"]`).attr('old_name', element.name);
				$(`#modal-empl-per [data-fieldvalue="${element.section}"]`).val(element.permission).formSelect();
			});	
		}
	});	
}

$(document).ready(function(){	
	$('#modal-empl-per #btn_save_permission').click(function(){				
		var per_role_name = $('#modal-empl-per #per_role_name').val();		
		
        $.each(permission_fields, function (key, element) {			   		
			var url = '/api/resource/Section Permission';    
			var type = 'POST';		
			var old_name = $(`#modal-empl-per [data-fieldvalue="${element}"]`).attr('old_name');
			if(old_name != ''){
				url += '/' + old_name;
				type = 'PUT';
			}	
			var permission = $(`#modal-empl-per [data-fieldvalue="${element}"]`).val();  			
			var args = {
				section: element,
				role: per_role_name,
				permission: permission
			};     
			frappe.ajax({
				type: type,
				url: url,
				args: args,
				callback: function (r) {
					if (!r.exc_type) {                
						
					}
				}
			}); 		
			//location.reload();				
			$('#modal-empl-per').modal('close');
        });    
	});
});

File: erpx_hrm/www/hr/grp_settings.py
import frappe
import json

def get_context(context):
    if frappe.session.user == 'Guest':
        frappe.local.flags.redirect_location = '/'
        raise frappe.Redirect

    context.user = frappe.session.user
    context.csrf_token = frappe.sessions.get_csrf_token()

    #Role list
    role_arr = ["Employee", "Expense Verified", "Expense Approver", "HR Manager", "Leave Approver"]
    role_list = []    
    for i_role in role_arr:                        
        user_list = frappe.db.sql_list("""
							select u.name from `tabUser` u
                            left join `tabHas Role` r ON  r.parent = u.name and u.name not in ('Administrator', 'Guest')
                            where r.role = %(role_name)s""", {"role_name": i_role})                  
        role_list.append({
            "name": i_role,
            "total_user": len(user_list) or 0,
            "user_list": user_list
        })            
    context.role_list = role_list

    #User list
    context.all_user = frappe.db.sql_list("""
							select name from `tabUser` where name not in ('Administrator', 'Guest')                            
                            """)  

    #Permission list
    context.permission_list = ["No Permission", "All Employees view only", "All Employees view and edit", "My view only", "My view and edit"]
    context.permission_section_list = [
        "Basic Information", "All About Me", "Certificate and License", "Compensation and Bank Information", "Compensation History",
        "Disciplinary Action", "Emergency Contact", "Family Information", "Personal Information", "Qualifications"
    ]
    return context